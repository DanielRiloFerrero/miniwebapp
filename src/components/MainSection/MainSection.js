import React, { useState } from 'react'
import {
  Row,
  Container,
  InputGroup,
  InputGroupAddon,
  Input,
  Button,
} from 'reactstrap';

import Loading from '../Loading/Loading'
import FilmCard from '../FilmCard/FilmCard'

import styles from './MainSection.module.scss'

import { getFilms } from '../../_assets/js/services'

const MainSection = () => {
  const [list, setList] = useState([])
  const [loading, setLoading] = useState(false)
  const [error, setError] = useState('')

  const handleClickSearchFilms = async () => {
    setLoading(true)
    setError('')
    setList([])

    const searhString = document.querySelector('#search').value
    if (searhString.length === 0) {
      setError('Ningún valor para hacer la búsqueda')
      setLoading(false)
      return
    }

    const ok = await getFilms(searhString)
    if (ok.length === 0) {
      setError('No se encontró ninguna película')
    } else {
      setList(ok)
    }
    setLoading(false)
  }

  return (
    <section className={styles.wrapper}>
      <Container>
        <InputGroup>
          <Input id='search' />
          <InputGroupAddon addonType='append'><Button onClick={() => handleClickSearchFilms()}>Buscar</Button></InputGroupAddon>
        </InputGroup>
      </Container>

      <Container>
        {error && <div>{error}</div>}
        {list.length > 0 && (<Row>{list.map(el => <FilmCard key={el.id} data={el} />)}</Row>)}
      </Container>

      {loading && <Loading />}
    </section>
  )
}

export default MainSection