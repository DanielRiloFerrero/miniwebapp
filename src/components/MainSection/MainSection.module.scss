@import '../../_assets/styles/abstracts/vars.scss';

.wrapper {
  padding: $p-s 0;
}