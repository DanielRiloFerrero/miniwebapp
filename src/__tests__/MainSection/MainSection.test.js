import React from "react"
import { render, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import MainSection from "../../components/MainSection/MainSection"

describe('MainSection tests', () => {
  it("render", () => {
    const { container } = render(<MainSection />);
    expect(container).toMatchSnapshot()
  })
})