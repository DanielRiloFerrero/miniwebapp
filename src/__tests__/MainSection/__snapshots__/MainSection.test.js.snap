// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainSection tests render 1`] = `
<div>
  <section
    class="wrapper"
  >
    <div
      class="container"
    >
      <div
        class="input-group"
      >
        <input
          class="form-control"
          id="search"
          type="text"
        />
        <div
          class="input-group-append"
        >
          <button
            class="btn btn-secondary"
            type="button"
          >
            Buscar
          </button>
        </div>
      </div>
    </div>
    <div
      class="container"
    >
      
    </div>
  </section>
</div>
`;
